### PR TITLE
Update loading logic

### PR DESCRIPTION
**Pull request by dobesv**
- Add 3 second timeout, show default variant if Google Optimize fails to
load (e.g. due to ad blocking)
- Add check for `typeof window !== 'undefined'`, show default variant
for SSR
- Use `window.google_optimize` to check if script is already loaded
- Don't rely on `dataLayer.hide.end` being present to decide whether to
wait

Fixes

* https://github.com/hudovisk/react-optimize/issues/9
* #8 
* #6

### CODE REVIEW

File: src/Experiment.js [hunk 1]
```diff
@@ -3,7 +3,8 @@ import OptimizeContext from "./OptimizeContext";
 
 class Experiment extends React.Component {
   static defaultProps = {
-    loader: null
+    loader: null,
+    timeout: 3000
```
Comment (tlaak) [inline src/Experiment.js]: Could this be configurable? It feels a bit too long.

Comment (dobesv) [reply src/Experiment.js]: Yes, it's a prop so just pass whatever you want to the `<Experiment ... />`

If you want to configure it globally you could `Experiment.defaultProps.timeout = 1000` during app startup.

**Comment by hudovisk** [comment]
Thanks for the PR, I rebase it to include some minor modifications. It is released in version [2.2.0](https://github.com/hudovisk/react-optimize/releases/tag/v2.2.0).

@all-contributors please add dobesv for code and docs, also add tiak for review.

**Comment by hudovisk** [comment]
@all-contributors 

please add dobesv for code and docs

please add tiak for review.

**Comment by hudovisk** [comment]
@all-contributors please add dobesv for code and doc

**Comment by hudovisk** [comment]
@all-contributors please add tiak for review

**Comment by hudovisk** [comment]
@all-contributors please add tiaak for review

**Comment by hudovisk** [comment]
@all-contributors please add tlaak for review

**Comment by hudovisk** [comment]
Sorry for the spam guys hahahaha first time using this bot

**Pull request closed without merging**